service_container/3.3-di-changes.rst

```diff
@@ -0,0 +1,747 @@
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            # default configuration for services in *this* file
+            _defaults:
+                # automatically injects dependencies in your services
+                autowire: true
+                # automatically registers your services as commands, event subscribers, etc.
+                autoconfigure: true
+                # this means you cannot fetch services directly from the container via $container->get()
+                # if you need to do this, you can override this setting on individual services
+                public: false
+
+            # makes classes in src/AppBundle available to be used as services
+            # this creates a service per class whose id is the fully-qualified class name
+            AppBundle\:
+                resource: '../../src/AppBundle/*'
+                # you can exclude directories or files
+                # but if a service is unused, it's removed anyway
+                exclude: '../../src/AppBundle/{Entity,Repository}'
+
+            # controllers are imported separately to make sure they're public
+            # and have a tag that allows actions to type-hint services
+            AppBundle\Controller\:
+                resource: '../../src/AppBundle/Controller'
+                public: true
+                tags: ['controller.service_arguments']
+
+            # add more services, or override services that need manual wiring
+            # AppBundle\Service\ExampleService:
+            #     arguments:
+            #         $someArgument: 'some_value'
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <defaults autowire="true" autoconfigure="true" public="false" />
+
+                <prototype namespace="AppBundle\" resource="../../src/AppBundle/*" exclude="../../src/AppBundle/{Entity,Repository}" />
+
+                <prototype namespace="AppBundle\Controller" resource="../../src/AppBundle/Controller" public="true">
+                    <tag name="controller.service_arguments" />
+                </prototype>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+
+        // _defaults and loading entire directories is not possible with PHP configuration
+        // you need to define your services one-by-one
+        use AppBundle\Controller\DefaultController;
+
+        $container->autowire(DefaultController::class)
+            ->setAutoconfigured(true)
+            ->addTag('controller.service_arguments')
+            ->setPublic(false);
+
```

Comment:
Shouldn't this be `true`? Like in the yaml config `public: true` under `AppBundle\Controller\`